swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1304 @@
+        interparticle distance.
+
+    self.redist_freq: int
+        particle redistribution frequency.
+        Redistribute a handful of particles every ``self.redistribute_frequency``
+        iteration. How many particles are redistributed is controlled with the
+        ``self.redistribute_frac`` parameter.
+
+    self.redist_frac: float
+        fraction of particles to be redistributed when doing so.
+
+    self.redist_reduct: float
+        multiply the ``self.redistribute_fraction`` parameter by this factor every
+        time particles are being redistributed. In certain difficult cases this
+        might help the generation to converge if set to < 1.
+
+    self.redist_stop: int
+        don't redistribute particles after this many iterations.
+
+    self.dumpfreq: int
+        frequency of dumps of the current state of the iteration. If set to zero,
+        no intermediate results will be stored.
+
+    self.dump_basename: str
+        Basename for intermediate dumps. The filename will be constructed as
+        ``basename + <5 digit zero padded iteration number> + .hdf5``
+
+    self.random_seed: int, optional
+        set a specific random seed
+
+
+
+    Methods
+    ----------
+
+    self.set_random_seed(seed: int)
+        set the random seed to whatever pleases you.
+
+    """
+    # iteration parameters
+    iter_max: int = 10000
+    iter_min: int = 0
+    converge_thresh: float = 1e-4
+    tolerance_part: float = 1e-3
+    displ_thresh: float = 1e-3
+    delta_init: Union[float, None] = None
+    delta_reduct: float = 1.0
+    delta_min: float = 1e-6
+    redist_freq: int = 20
+    redist_frac: float = 0.01
+    redist_reduct: float = 1.0
+    redist_stop: int = 200
+    dump_basename: str = "IC-generation-iteration-"
+    dumpfreq: int = 50
+
+    def __init__(self):
+        # always set one random seed.
+        self.set_random_seed(666)
+        return
+
+    def set_random_seed(self, seed: int = 666):
+        r"""
+        Set the random seed to whatever pleases you.
+
+        Parameters
+        --------------
+
+        seed: int, optional
+            seed to use.
+        """
+        np.random.seed(seed)
+        return
+
+
+class IterData(object):
+    r"""
+    contains data relevant during the iteration. This data is
+    automatically set up, and no user is supposed to tinker
+    with this data during/before the iteration.
+
+    Attributes
+    -------------
+
+    self.mean_interparticle_distance: float
+        the mean interparticle distance
+
+    self.delta_r_norm: float
+        norm to be used to compute the displacement "force"
+
+    self.delta_r_norm_min: float
+        lower threshold for "force" normalisation.
+
+    self.compute_delta_r_norm: bool
+        whether the user wants the code to compute an appropriate
+        normalisation.
+
+    self.Nngb: float
+        number of neighbours to be searched for
+
+    self.Nngb_int: int
```

Comment:
Let's just round `self.Nngb` up for this.

Reply:
Done.